extra/utils.py

```diff
@@ -1,5 +1,11 @@
 # type: ignore
 import pickle
+from hashlib import md5
+from zipfile import ZipFile
```

Comment:
Ugh, this is not noticeably faster. Readability is top priority!

Reply:
True, taste. Reverted and grouped the imports into one line.